components/identifier-generator/back/src/Domain/Model/Property/Process.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Automation\IdentifierGenerator\Domain\Model\Property;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @phpstan-type ProcessOperator 'EQUALS'|'LOWER_OR_EQUAL_THAN'
+ * @phpstan-type ProcessNormalized array{type: 'no'|'truncate'|'nomenclature', operator?: string, value?: int}
+ */
+final class Process
+{
+    public const PROCESS_TYPE_NO = 'no';
+    public const PROCESS_TYPE_TRUNCATE = 'truncate';
+    public const PROCESS_TYPE_NOMENCLATURE = 'nomenclature';
+
+    public const PROCESS_OPERATOR_EQ = 'EQUALS';
+    public const PROCESS_OPERATOR_LTE = 'LOWER_OR_EQUAL_THAN';
+
+    public function __construct(
```

Comment:
Same, you have only 1 place in the code with new Process, and it's in a test ; you should set this private.
```suggestion
    private function __construct(
```